client/blocks/site/index.jsx

```diff
@@ -88,7 +88,7 @@ export default React.createClass( {
 		return (
 			<div className={ siteClass }>
 				<a className="site__content"
-					href={ this.props.homeLink ? site.URL : this.props.href }
+					href={ this.props.homeLink ? site.slug : this.props.href }
```

Comment:
Maybe we need to assign `URL` as the `options.unmapped_url` when determining that the site is either a redirect or has a conflict?

https://github.com/Automattic/wp-calypso/blob/833c11a/client/lib/site/computed-attributes.js#L32-L37